commerce_coordinator/apps/core/models.py

```diff
@@ -36,3 +40,56 @@ def get_full_name(self):
 
     def __str__(self):
         return str(self.get_full_name())
+    
+    def add_lms_user_id(self, called_from):
+        """
+        If this user does not already have an LMS user id, look for the id in social auth. If the id can be found,
+        add it to the user and save the user.
+        The LMS user_id may already be present for the user. It may have been added from the jwt (see the
+        EDX_DRF_EXTENSIONS.JWT_PAYLOAD_USER_ATTRIBUTE_MAPPING settings) or by a previous call to this method.
+        Arguments:
+            called_from (String): Descriptive string describing the caller. This will be included in log messages.
+        """
+        if not self.lms_user_id:
+            # Check for the LMS user id in social auth
+            lms_user_id_social_auth, social_auth_id = self._get_lms_user_id_from_social_auth()
+            if lms_user_id_social_auth:
+                self.lms_user_id = lms_user_id_social_auth
+                self.save()
+                log_message = (
+                    'Saving lms_user_id from social auth with id %s '
+                    'for user %s. Called from %s',
+                    social_auth_id,
+                    self.id,
+                    called_from
+                )
+                logger.info(log_message)
+            else:
+                log_message = (
+                    'No lms_user_id found in social auth with id %s '
+                    'for user %s. Called from %s',
+                    social_auth_id,
+                    self.id,
+                    called_from
+                )
```

Comment:
`social_auth_id` won't exist if no record is found in the social auth table.

```suggestion
                log_message = f'No lms_user_id found in social auth for user {self.id}. Called from {called_from}'
```